ui/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentService.scala

```diff
@@ -1,134 +1,153 @@
-import pl.touk.nussknacker.ui.process.repository.FetchingProcessRepository
-import pl.touk.nussknacker.ui.process.subprocess.{SubprocessDetails, SubprocessRepository}
-import pl.touk.nussknacker.ui.process.{ConfigProcessCategoryService, ProcessObjectsFinder}
+import pl.touk.nussknacker.ui.process.{ConfigProcessCategoryService, ProcessObjectsFinder, ProcessService}
 import pl.touk.nussknacker.ui.security.api.{LoggedUser, NussknackerInternalUser}
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContext, Future}
 
 trait ComponentService {
   def getComponentsList(user: LoggedUser): Future[List[ComponentListElement]]
+
+  def getComponentUsages(componentId: ComponentId)(implicit user: LoggedUser): Future[XError[List[ComponentUsagesInScenario]]]
 }
 
 object DefaultComponentService {
+
   import WrongConfigurationAttribute._
-  import pl.touk.nussknacker.engine.component.ComponentsUiConfigExtractor.ComponentsUiConfig
+
+  type ComponentIdProviderWithError = Validated[List[ComponentWrongConfiguration[_]], ComponentIdProvider]
 
   def apply(config: Config,
             processingTypeDataProvider: ProcessingTypeDataProvider[ProcessingTypeData],
-            fetchingProcessRepository: FetchingProcessRepository[Future],
-            subprocessRepository: SubprocessRepository,
+            processService: ProcessService,
             categoryService: ConfigProcessCategoryService)(implicit ec: ExecutionContext): DefaultComponentService = {
-    val wrongConfigurations = findWrongConfigurations(processingTypeDataProvider, categoryService)
+    val futureComponentIdProvider = prepareComponentProvider(processingTypeDataProvider, processService, categoryService)
+    val componentIdProvider = Await.result(futureComponentIdProvider, 30 seconds)
 
-    if (wrongConfigurations.nonEmpty)
-      throw ComponentConfigurationException(s"Wrong configured components were found.", wrongConfigurations)
-    else
-      new DefaultComponentService(config, processingTypeDataProvider, fetchingProcessRepository, subprocessRepository, categoryService)
+    componentIdProvider
+      .map(new DefaultComponentService(config, processingTypeDataProvider, processService, categoryService, _))
+      .valueOr(wrongConfigurations => throw ComponentConfigurationException(s"Wrong configured components were found.", wrongConfigurations))
   }
 
-  private def findWrongConfigurations(processingTypeDataProvider: ProcessingTypeDataProvider[ProcessingTypeData], categoryService: ConfigProcessCategoryService) = {
-    val components = processingTypeDataProvider.all.flatMap{
+  private def prepareComponentProvider(processingTypeDataProvider: ProcessingTypeDataProvider[ProcessingTypeData], processService: ProcessService, categoryService: ConfigProcessCategoryService)(implicit ec: ExecutionContext): Future[ComponentIdProviderWithError] = {
```

Comment:
This should not use processService and should not return Future - you added throwing an exception if someone tries to override fragment componentId, so you can pass empty set in line 88 :)